### PR TITLE
wip(kafka): switch out `kafka-python` with `confluent-kafka`

### PR DESCRIPTION
**Pull request by hazzadous**
This is to enable a few things:

 1. pulling out metrics from librdkafka. This can be done to some extent
    from kafka-python, but getting the "standard" onces from librdkafka
    would be ideal. I want to get these in such that we can see the
    details on where we might drop events.
 2. support for `enable.idempotence` such that we can emit events
    without duplicates, which will simplify e.g. handling in session
    recordings.
 3. support for Kafka Transaction API such that we can commit multiple
    events at once, thereby simplifying the handling of multievent
    processing down the pipeline.

Separately, it would give more confidence in the producer to be on
librdkafka which is arguably more battle tested than `kafka-python`

This is a WIP. I'm going to split out the django-prometheus part as a separate PR as there is some detail in getting the exporter working with multiple processes.

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: posthog/kafka_client/client.py [hunk 1]
```diff
@@ -87,12 +93,28 @@ def __init__(self, test=TEST):
...
+                }
             )
 
+        self._cancelled = False
+        self._poll_thread = Thread(target=self._poll_loop)
+        self._poll_thread.start()
+
+    def _poll_loop(self):
+        while not self._cancelled:
+            self.producer.poll(0.1)
```
Comment (guidoiaquinti) [inline posthog/kafka_client/client.py:112]: I was a bit skeptical about this but it's the same setting used by the [asyncio example](https://github.com/confluentinc/confluent-kafka-python/blob/master/examples/asyncio_example.py#L47) from upstream

File: posthog/kafka_client/client.py [hunk 2]
```diff
@@ -104,10 +126,17 @@ def produce(self, topic: str, data: Any, key: Any = None, value_serializer: Opti
         b = value_serializer(data)
         if key is not None:
             key = key.encode("utf-8")
-        self.producer.send(topic, value=b, key=key)
+        
+        result = Future()
 
-    def close(self):
-        self.producer.flush()
+        def ack(err, msg):
```
Comment (guidoiaquinti) [inline posthog/kafka_client/client.py:132]: Is this section thread safe?

File: requirements.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 #
-# This file is autogenerated by pip-compile with python 3.8
+# This file is autogenerated by pip-compile with python 3.9
```
Comment (guidoiaquinti) [inline requirements.txt:2]: Nit: we use Python 3.8

Comment (hazzadous) [reply requirements.txt:2]: Damn it sorry

**Comment by frankh** [comment]
💩

**Pull request closed without merging**